lib_standard_app/main.c

```diff
@@ -80,6 +89,76 @@ WEAK __attribute__((section(".boot"))) int main() {
     }
 
     app_exit();
+}
+
+#ifdef HAVE_SWAP
+static void library_app_main_helper(struct libargs_s *args) {
+    PRINTF("Inside library\n");
+    switch (args->command) {
+        case SIGN_TRANSACTION:
+            if (swap_copy_transaction_parameters(args->create_transaction)) {
+                common_app_init();
+
+                G_called_from_swap = true;
+                G_swap_response_ready = false;
+
+                app_main();
+            }
+            break;
+        case CHECK_ADDRESS:
+            // ensure result is zero if an exception is thrown
+            args->check_address->result = 0;
+            args->check_address->result = swap_handle_check_address(args->check_address);
+            break;
+        case GET_PRINTABLE_AMOUNT:
+            swap_handle_get_printable_amount(args->get_printable_amount);
+            break;
+        default:
+            break;
+    }
+}
+
+static void library_app_main(struct libargs_s *args) {
+    BEGIN_TRY {
+        TRY {
+            library_app_main_helper(args);
+        }
+        CATCH_OTHER(e) {
+            PRINTF("Exception caught %d\n", e);
+        }
+        FINALLY {
+            os_lib_end();
+        }
+    }
+    END_TRY;
+}
+#endif // HAVE_SWAP
+
+__attribute__((section(".boot"))) int main(int arg0) {
+    // exit critical section
+    __asm volatile("cpsie i");
+
+    // ensure exception will work as planned
+    os_boot();
+
+#ifdef HAVE_SWAP
+    if (arg0 == 0) {
+        // called from dashboard as standalone Stellar App
+        standalone_app_main();
+    } else {
+        // Called as library from another app
+        libargs_t *args = (libargs_t *) arg0;
+        if (args->id == 0x100) {
```

Comment:
It would be nice to have the real define behind the `0x100` value, but I have no idea what it actually is